docs/articles/documentation/using-testcafe/configuration-file.md

```diff
@@ -0,0 +1,715 @@
+
+### filter.fixtureGrep
+
+Runs tests whose names match the specified `grep` pattern.
+
+```json
+{
+    "filter": {
+        "fixtureGrep": "Page.*"
+    }
+}
+```
+
+*CLI*: [-F, --fixture-grep](command-line-interface.md#-f-pattern---fixture-grep-pattern)  
+*API*: [runner.filter](programming-interface/runner.md#filter)
+
+### filter.testMeta
+
+Runs tests whose [metadata](../test-api/test-code-structure.md#specifying-testing-metadata) [matches](https://lodash.com/docs/#isMatch) the specified key-value pair.
+
+```json
+{
+    "filter": {
+        "testMeta": {
+            "device": "mobile",
+            "env": "production"
+        }
+    }
+}
+```
+
+This configuration runs tests whose metadata's `device` property is set to `mobile`, and `env` property is set to `production`.
+
+*CLI*: [--test-meta](command-line-interface.md#--test-meta-keyvaluekey2value2)  
+*API*: [runner.filter](programming-interface/runner.md#filter)
+
+### filter.fixtureMeta
+
+Runs tests whose fixture's [metadata](../test-api/test-code-structure.md#specifying-testing-metadata) [matches](https://lodash.com/docs/#isMatch) the specified key-value pair.
+
+```json
+{
+    "filter": {
+        "fixtureMeta": {
+            "device": "mobile",
+            "env": "production"
+        }
+    }
+}
+```
+
+This configuration runs tests whose fixture's metadata has the `device` property set to `mobile` and the `env` property set to the `production`.
+
+*CLI*: [--fixture-meta](command-line-interface.md#--fixture-meta-keyvaluekey2value2)  
+*API*: [runner.filter](programming-interface/runner.md#filter)
+
+## appCommand
+
+Executes the specified shell command before running tests.
+
+```json
+{
+    "appCommand": "node server.js"
+}
+```
+
+Use the `appCommand` property to launch the application you are going to test. This application is automatically terminated after testing is finished.
+
+The [appInitDelay](#appinitdelay) property specifies the amount of time allowed for this command to initialize the tested application.
+
+> TestCafe adds `node_modules/.bin` to `PATH` so that you can use the binaries the locally installed dependencies provide without prefixes.
+
+*CLI*: [-a, --app](command-line-interface.md#-a-command---app-command)  
+*API*: [runner.startApp](programming-interface/runner.md#startapp)
+
+## concurrency
+
+Specifies the number of browser instances that should run tests concurrently.
+
+```json
+{
+    "concurrency": 3
+}
+```
+
+TestCafe opens several instances of the same browser and creates a pool of browser instances. Tests are run concurrently against this pool, that is, each test is run in the first free instance.
+
+See [Concurrent Test Execution](common-concepts/concurrent-test-execution.md) for more information about concurrent test execution.
+
+*CLI*: [-c, --concurrency](command-line-interface.md#-c-n---concurrency-n)  
+*API*: [runner.concurrency](programming-interface/runner.md#concurrency)
+
+## debugOnFail
```

Comment:
`debugOnFail` probably should be closer to `debugMode` and `appInitDelay` to `appCommand`